lib/ex_sdp/origin.ex

```diff
@@ -12,16 +12,18 @@ defmodule ExSDP.Origin do
   alias ExSDP.ConnectionData
 
   @enforce_keys [
+    :username,
     :session_id,
     :session_version,
     :address
   ]
-  defstruct @enforce_keys ++ [:username]
+  defstruct @enforce_keys
 
   @type t :: %__MODULE__{
-          username: binary() | nil,
+          username: binary(),
           session_id: binary(),
           session_version: binary(),
+          # FIXME this cannot be ConnectionData.sdp_address() as it has no ttl field for IPv4
```

Comment:
TTL is optional in `ConnectionData.IP4` but origin in SDP cannot have TTL so in fact we shouldn't use `ConnectionData.sdp_addresses()` here

Reply:
I think it can:
> Sessions using an IPv4 multicast connection address MUST also have
      a time to live (TTL) value present in addition to the multicast
      address

https://tools.ietf.org/html/rfc4566#section-5.7

Reply:
Yeah, that's true but I woud move removing these structs to new PR too cuz it will involve some more changes.